pallets/ocex/src/tests.rs

```diff
@@ -20,113 +20,605 @@ use frame_support::{
+}
+
+
+fn mint_into_account(account_id: AccountId32){
+	Balances::deposit_creating(&account_id, 100000000000000);
+}
+
+fn create_asset_and_credit(asset_id: u128, account_id: AccountId32){
+	assert_ok!(
+		Assets::create(
+			Origin::signed(account_id.clone().into()),
+			asset_id.into(),
+			account_id.clone().into(),
+			100_u128		
+		)
+	);
+}
+
+fn create_account_id() -> AccountId32{
+	const PHRASE: &str =
+		"news slush supreme milk chapter athlete soap sausage put clutch what kitten";
+	let keystore = KeyStore::new();
+	let account_id: AccountId32 = SyncCryptoStore::sr25519_generate_new(
+		&keystore,
+		KEY_TYPE,
+		Some(&format!("{}/hunter1", PHRASE)),
+	)
+	.expect("Unable to create sr25519 key pair")
+	.try_into()
+	.expect("Unable to convert to AccountId32");
+
+	return account_id;
+}
+
+fn create_public_key() -> sp_core::sr25519::Public {
+	const PHRASE: &str =
+		"news slush supreme milk chapter athlete soap sausage put clutch what kitten";
+	let keystore = KeyStore::new();
+	let account_id = SyncCryptoStore::sr25519_generate_new(
+		&keystore,
+		KEY_TYPE,
+		Some(&format!("{}/hunter1", PHRASE)),
+	)
+	.expect("Unable to create sr25519 key pair");
+
+	return account_id
+}
+
+
+fn create_mmr_with_one_account() -> H256{
+	let account_id = create_account_id();
+	let mut snapshot: BTreeMap<AccountId, AccountInfo<AccountId, Balance, ProxyLimit>> = Default::default();
+	assert_ok!(OCEX::register_main_account(Origin::signed(account_id.clone().into()), account_id.clone().into()));
+	let account_info = Accounts::<Test>::get::<AccountId32>(account_id.clone().into()).unwrap();
+	snapshot.insert(account_id.clone().into(), account_info.clone().into());
+	calculate_mmr_root(&mut snapshot.values()).unwrap()
+}
+
+#[derive(Eq, PartialEq, Clone, Debug, Default)]
+pub struct MergeAccountInfo(pub [u8; 32]);
+impl Merge for MergeAccountInfo {
+    type Item = MergeAccountInfo;
+    fn merge(lhs: &Self::Item, rhs: &Self::Item) -> Result<Self::Item, ckb_merkle_mountain_range::Error> {
+        let mut bytes = Vec::<u8>::with_capacity(64);
+        lhs.0.map(|byte| bytes.push(byte));
+        rhs.0.map(|byte| bytes.push(byte));
+        Ok(MergeAccountInfo(sp_core::blake2_256(&bytes)))
+    }
+}
+
+/// Calculates the MMR root for a given vector of accounts
+pub fn calculate_mmr_root(
+    accounts: &mut Values<AccountId, AccountInfo<AccountId, Balance, ProxyLimit>>,
+) -> anyhow::Result<H256> {
+    let store = MemStore::default();
+    let mut mmr = MMR::<_, MergeAccountInfo, _>::new(0, &store);
+    accounts.by_ref().for_each(|value| {
+        let bytes = value.encode();
+        if let Err(err) = mmr.push(MergeAccountInfo(sp_core::blake2_256(&bytes))) {
+            log::error!(target: "mmr", "Unable to push account into MMR calculator: {:?}", err);
+        }
+    });
+
+    match mmr.get_root() {
+        Ok(root) => Ok(H256::from(root.0)),
+        Err(err) => Err(anyhow::Error::msg(format!("unable to calculate MMR root: {:?}", err))),
+    }
+}
```

Comment:
No separate test for `insert_enclave`.

Reply:
It was commented that `insert_enclave` is only used for testing purpose, Ideally I guess we would be calling `register_enclave`

Reply:
then at least it should be annotated wit `#[doc(hidden)]`

Reply:
Updated ✅